doc/md/guides/orms/beego.md

```diff
@@ -0,0 +1,305 @@
+---
+id: beego
+title: Automatic migration planning for Beego
+slug: /guides/orms/beego
+---
+
+## TL;DR
+* [Beego](https://github.com/beego/beego) is an open-source web framework that's widely used in the Go community.
+* [Atlas](https://atlasgo.io) is an open-source tool for inspecting, planning, linting and
+  executing schema changes to your database.
+* Developers using Beego can use Atlas to automatically plan schema migrations
+  for them, based on the desired state of their schema instead of crafting them by hand.
+
+## Automatic migration planning for Beego
+
+Beego is a popular web framework widely used in the Go community. Among many other features,
+Beego provides an ORM that allows work with popular databases such as MySQL, PostgreSQL, SQLite3, and more.
+
+Beego allows developers to manage their database schemas using its [syncdb](https://github.com/beego/beedoc/blob/master/en-US/mvc/model/cmd.md#database-schema-generation)
+feature, which is usually sufficient during development and in many simple cases. 
+
+However, at some point, teams need more control and decide to employ 
+the [versioned migrations](/concepts/declarative-vs-versioned#versioned-migrations) 
+methodology. Once this happens, the responsibility for planning migration scripts and making
+sure they are in line with what Beego expects at runtime is moved to developers.
+
+Atlas can automatically plan database schema migrations for developers using Beego.
+Atlas plans migrations by calculating the diff between the _current_ state of the database,
+and its _desired_ state.
+
+In the context of versioned migrations, the current state can be thought of as the database schema that would have
+been created by applying all previous migration scripts.
+
+The desired schema of your application can be provided to Atlas via an [External Schema Datasource](/atlas-schema/projects#data-source-external_schema)
+which is any program that can output a SQL schema definition to stdout.
+
+To use Atlas with Beego, users can utilize the [Beego Atlas Provider](https://github.com/ariga/atlas-provider-beego),
+a small Go program that can be used to load the schema of a Beego project into Atlas.
+
+In this guide, we will show how Atlas can be used to automatically plan schema migrations for
+Beego users.
+
+## Prerequisites
+
+* A local [Beego](https://github.com/beego/beego) project.
+
+## Using the Beego Atlas Provider 
+
+In this guide, we will use the [Beego Atlas Provider](https://github.com/ariga/atlas-provider-beego)
+to automatically plan schema migrations for a Beego project.
+
+### Installation
+
+Install Atlas from macOS or Linux by running:
+```bash
+curl -sSf https://atlasgo.sh | sh
+```
+See [atlasgo.io](https://atlasgo.io/getting-started#installation) for more installation options.
+
+Install the provider by running:
+```bash
+go get -u ariga.io/atlas-provider-beego
```

Comment:
```suggestion
go install ariga.io/atlas-provider-beego@latest
```

Reply:
I still think this is the right way for the standalone mode because the user will not depend on `atlas-provider-beego` in their code base. And usage of `go run -mod` will cause their `go.sum`/`go.mod` changes every time they run the atlas command.